flight-telemetry/src/main/java/com/hazelcast/jet/demo/FlightTelemetry.java

```diff
@@ -183,34 +184,34 @@ private static Pipeline buildPipeline() {
                 ).setName("Calculate linear trend of aircraft"); // (timestamp, aircraft_id, aircraft_with_assigned_trend)
 
         // Filter ascending flights
-        StreamStage<TimestampedEntry<Long, Aircraft>> takingOffFlights = flights
+        StreamStage<KeyedWindowResult<Long, Aircraft>> takingOffFlights = flights
                 .filter(e -> e.getValue().getVerticalDirection() == ASCENDING)
                 .setName("Filter ascending aircraft");
         // Write ascending flights to an IMap
         takingOffFlights.drainTo(Sinks.map(TAKE_OFF_MAP)); // (aircraft_id, aircraft)
 
         //Filter descending flights
-        StreamStage<TimestampedEntry<Long, Aircraft>> landingFlights = flights
+        StreamStage<KeyedWindowResult<Long, Aircraft>> landingFlights = flights
                 .filter(e -> e.getValue().getVerticalDirection() == DESCENDING)
                 .setName("Filter descending aircraft");
         // Write descending flights to an IMap
         landingFlights.drainTo(Sinks.map(LANDING_MAP)); // (aircraft_id, aircraft)
 
         // Enrich aircraft with the noise info and calculate max noise
         // in 60secs windows sliding by 30secs.
-        StreamStage<TimestampedEntry<String, Integer>> maxNoise = flights
+        StreamStage<KeyedWindowResult<String, Integer>> maxNoise = flights
                 .map(e -> entry(e.getValue(), getNoise(e.getValue())))
                 .setName("Enrich with noise info")// (aircraft, noise)
                 .window(slidingWindow)
                 .groupingKey(e -> e.getKey().getAirport() + "_AVG_NOISE")
-                .aggregate(maxBy(comparingInt(Entry::getValue)),
-                        (winStart, winEnd, key, e) -> new TimestampedEntry<>(winEnd, key, e.getValue()))
+                .aggregate(maxBy(ComparatorEx.comparingInt(Entry::getValue)))
+                .map(res -> new KeyedWindowResult<>(res.start(), res.end(), res.key(), res.result().getValue()))
```

Comment:
I think you could use `AggregateOperation.andThen` here instead

Reply:
👍